persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq

```diff
@@ -76,6 +76,96 @@ UPDATE Conversation
 SET last_read_date = :last_read_date
 WHERE qualified_id = :qualified_id;
 
+CREATE VIEW IF NOT EXISTS ConversatonsList AS
+SELECT
+Conversation.qualified_id AS qualifiedId,
+CASE (Conversation.type)
+    WHEN "ONE_ON_ONE" THEN  User.name
+    WHEN "CONNECTION_PENDING" THEN connection_user.name
+    ELSE Conversation.name
+    END AS name,
+Conversation.type,
+Call.status AS callStatus,
+CASE (Conversation.type)
+    WHEN "ONE_ON_ONE" THEN  User.preview_asset_id
+    WHEN "CONNECTION_PENDING" THEN connection_user.preview_asset_id
+    END AS previewAssetId,
```

Comment:
I'd suggest pulling in the entire connection.user instead of extracting most fields individually. This would also simply the mapping later and make it less error prone since you can re-use the user mapper.

Reply:
But then most of the fields are redundant.

Reply:
You pull in almost all fields. I don't think the overhead is that big.

Reply:
it's not about the data, because we have all the data we need, what I'm doing there instead of having connection userId, myUserId, otherUserid -> always return userId, and based on the type of the conversation put the correct value in that!

Reply:
And I say. Pull connection.user as connectedUser if type is oneToOne and deal with populating the right fields in the mapper. Less logic in the query.

Reply:
@mchenani @typfel we will address this in another PR. For the sake of not holding other code off we'll merge as is for now.